source/h1_connection.c

```diff
@@ -437,21 +473,12 @@ struct aws_http_stream *s_make_request(
 
     /* Insert new stream into pending list, and schedule outgoing_stream_task if it's not already running. */
     int new_stream_error_code = AWS_ERROR_SUCCESS;
-    bool should_schedule_task = false;
 
     { /* BEGIN CRITICAL SECTION */
         s_h1_connection_lock_synced_data(connection);
-
         if (connection->synced_data.new_stream_error_code) {
             new_stream_error_code = connection->synced_data.new_stream_error_code;
-        } else {
-            aws_linked_list_push_back(&connection->synced_data.new_client_stream_list, &stream->node);
-            if (!connection->synced_data.is_outgoing_stream_task_active) {
-                connection->synced_data.is_outgoing_stream_task_active = true;
-                should_schedule_task = true;
-            }
         }
-
         s_h1_connection_unlock_synced_data(connection);
     } /* END CRITICAL SECTION */
```

Comment:
Maybe we should move this whole critical section into `activate()`?
That way we only grab the lock once
I could see an argument that you'd rather see a "no new streams allowed" exception sooner than later
I'm really not sure on this one, just bringing it up.

Reply:
I'm not sure on this one either, but I suspect a fast fail if there's a connection error (such as a connection: close header received), we want this to fail sooner rather than later.